src/scripts/domain/moveCalculator.ts

```
@@ -225,11 +229,351 @@ export class MoveCalculator {
+    mainDir: RVector2,
+  ): boolean {
+    let cell: Wall | PieceInfo | null = null;
+
+    let cellsAdvanced = 0;
+
+    // While empty cell
+    while (
+      cell === null ||
+      (cell !== WallCell && (cell !== piece || toRemove.includes(cell)))
+    ) {
+      for (const dir of this.cardinalDirections) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)